be/src/column/chunk.h

```diff
@@ -31,6 +31,44 @@ class ChunkPB;
 namespace vectorized {
 
 class DatumTuple;
+class ChunkExtraData;
+using ChunkExtraDataPtr = std::shared_ptr<ChunkExtraData>;
+
+/**
+ * ChunkExtraData is an extra data which can be used to extend Chunk and 
+ * attach extra infos beside the schema. eg, In Stream MV scenes, 
+ * the hidden `_op_` column can be added in the ChunkExtraData.
+ *
+ * NOTE: `ChunkExtraColumnsData` is a specific implementation which includes extra 
+ * columns besides the schema, and it supports common methods like Chunk.
+ *
+ * TODO: Optimize old variables use this extendable method, eg owner_info.
+ */
+struct ChunkExtraDataMeta {
+    TypeDescriptor type;
+    bool is_null;
+    bool is_const;
+};
+class ChunkExtraData {
+public:
+    virtual ~ChunkExtraData() = default;
+    ChunkExtraData() = default;
+
+    virtual std::vector<ChunkExtraDataMeta> chunk_data_metas() const = 0;
+
+    virtual void filter(const Buffer<uint8_t>& selection) const = 0;
+    virtual void filter_range(const Buffer<uint8_t>& selection, size_t from, size_t to) const = 0;
+    virtual ChunkExtraDataPtr clone_empty(size_t size) const = 0;
+    virtual void append(const ChunkExtraDataPtr& src, size_t offset, size_t count) = 0;
```

Comment:
Yep. I've added two memory_usage methods, other methods are only used in MemTable or ChunkAggregator. 

This may be a problem for `ChunkExtraData ` because there are many methods in  Chunk class.  But this should be controlled by the specific uses of `ChunkExtraData ` (eg in stream mv, we can should check the usages of  `ChunkExtraData ` carefully).